attestation/docker/docker_test.go

```diff
@@ -0,0 +1,196 @@
+			},
+			validate: func(a *Attestor, err error, name string) {
+				require.Equal(t, a.ImageDigest[cryptoutil.DigestValue{Hash: crypto.SHA256}], "", "TestName: %s", name)
+				require.Equal(t, []string{""}, a.ImageReferences, "TestName: %s", name)
+				require.NoError(t, err, "TestName: %s", name)
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			a := New()
+
+			testProductSet := make(map[string]attestation.Product)
+			for _, prod := range tt.testProducts {
+				decoded, err := base64.StdEncoding.DecodeString(prod)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				hashes := []cryptoutil.DigestValue{{Hash: crypto.SHA256}}
+
+				prodDigest, err := cryptoutil.CalculateDigestSetFromBytes([]byte(decoded), hashes)
+				if err != nil {
+					t.Errorf("Failed to calculate digest set from bytes: %v", err)
+				}
+
+				file := SetupTest(t, prod)
+
+				testProductSet[file.Name()] = attestation.Product{
+					MimeType: "application/json",
+					Digest:   prodDigest,
+				}
+			}
+
+			tp := testproducter.TestProducter{}
+			tp.SetProducts(testProductSet)
+			ctx, err := attestation.NewContext("test", []attestation.Attestor{tp, a})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			err = ctx.RunAttestors()
+
+			tt.validate(a, err, tt.name)
+
+			for prod := range tp.Products() {
+				os.Remove(prod)
+			}
+		})
+	}
+}
+
+func SetupTest(t *testing.T, productFileData string) *os.File {
+	s := sha1.New()
+	s.Write([]byte(productFileData))
+	bs := s.Sum(nil)
+
+	file, err := os.CreateTemp("", fmt.Sprintf("%x.json", bs))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(productFileData)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = file.Write([]byte(decoded))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return file
+}
+
+func TestNew(t *testing.T) {
+	a := New()
+	if a.Name() != Name {
+		t.Errorf("expected Name to be %s, got %s", Name, a.Name())
```

Comment:
Can use the `assert` package everywhere `t.Error` is called to be consistent with our other tests